### PR TITLE
feat(rnd, infra): Add deployment for backend

### PR DESCRIPTION
**Pull request by aarushik93**
### Background

Deploying backend to k8s. This is the base structure and all other services (marketplace, builder etc) will follow a similar path.

### Changes 🏗️

Helm charts for server. 
Iams for server
Enable workload identtity & bind KSA with GSA

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
Enhancement, Configuration changes, Documentation

___

### **Description**
- Removed mounting of static files for frontend in `server.py`.
- Added environment variable for `PYRO_HOST` in `service.py`.
- Added IAM module and updated networking configurations in Terraform.
- Enabled workload identity and IP allocation policy in GKE cluster module.
- Created IAM resources for service accounts and bindings.
- Added Helm chart for AutoGPT server with various templates and values.
- Added development-specific values for Helm chart.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>server.py</strong><dd><code>Remove frontend static files mounting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/server/server.py

<li>Removed mounting of static files for frontend.<br> <li> Included router for application.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-94fec4bd2ec2c2e02a2e106e1ebcdce2d819c8c7fc0a30561d31fabb0e82d1ee">+0/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>service.py</strong><dd><code>Add environment variable for Pyro host configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/autogpt_server/autogpt_server/util/service.py

<li>Added environment variable for <code>PYRO_HOST</code>.<br> <li> Updated <code>PyroNameServer</code> and <code>__start_pyro</code> to use <code>PYRO_HOST</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-0688cfa91a411a53076603b85abc3541971cd2de1d2812b4f467ddfed1eea443">+5/-3</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>23 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>main.tf</strong><dd><code>Add IAM module and update networking configurations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/main.tf

- Added IAM module.
- Updated networking module with new variables.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-0ed6465c6e4b9f7ee3af0bc4cf5eb58d4c2cefa1ad156cc6ee969bfd82cf5820">+18/-5</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.tf</strong><dd><code>Enable workload identity and IP allocation policy</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/modules/gke_cluster/main.tf

<li>Enabled workload identity configuration.<br> <li> Added IP allocation policy for pods and services.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-89c8bcdf781ef0ef53fdf84c0043ad0cc42ef05eef4e2e1960140c27ca749603">+14/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.tf</strong><dd><code>Create IAM resources for service accounts and bindings</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/modules/iam/main.tf

<li>Created IAM resources for service accounts, workload identity <br>bindings, and role bindings.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-5e9b5924e857f74731b09da94f83dbc65a94b20dd4d85946640681bfd91390c7">+26/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>outputs.tf</strong><dd><code>Add output for service account emails</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/modules/iam/outputs.tf

- Added output for service account emails.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-9089fa8ddfd041873b6409da7f31088441ff2306fab120c101c4a845133aa875">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>variables.tf</strong><dd><code>Define variables for IAM configurations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/modules/iam/variables.tf

<li>Defined variables for project ID, service accounts, workload identity <br>bindings, and role bindings.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-098e09e028a6688b4930404decb4ffd1020cb3f1281c0c1cc2ad1b9eeea57330">+29/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.tf</strong><dd><code>Add secondary IP ranges for pods and services</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/modules/networking/main.tf

- Added secondary IP ranges for pods and services.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-1abe628a79ad15289ee40c3ce0fb7406beda15e6dcc043212c5e33ce4bcf3bfd">+10/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>variables.tf</strong><dd><code>Define variables for IP CIDR ranges</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/modules/networking/variables.tf

- Defined variables for pods and services IP CIDR ranges.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-f66cdcd16bea11d28230833373bfa8f22132d876771da96fcb37e34ac2c58a29">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.tf</strong><dd><code>Change resource type for static IPs</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/modules/static_ip/main.tf

<li>Changed resource type from <code>google_compute_address</code> to <br><code>google_compute_global_address</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-7c56c473b9e3d6a7477f206b2dcc0bc711852bd9d6df822fc7334051c5d63ed6">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>outputs.tf</strong><dd><code>Update outputs for static IP resource type change</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/modules/static_ip/outputs.tf

- Updated outputs to reflect the change in resource type.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-135a52383fa2d0b4887cb3bee2971832df6bb298a208affe09b1ce067ce95f0b">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>variables.tf</strong><dd><code>Add variables for IAM and networking configurations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/variables.tf

<li>Added variables for service accounts, workload identity bindings, role <br>bindings, and IP CIDR ranges.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-b6aaf3634e43bd484993882fcc30b1c1ffc70b0e21bee8e0d129bfb52a444f80">+37/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>.helmignore</strong><dd><code>Add .helmignore for Helm package exclusions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/.helmignore

<li>Added .helmignore file to exclude unnecessary files from Helm package.<br> <br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-5269a10cf5adde046f881247cca99fffd80f5d42e1f9c4966c4d6e264869cd96">+23/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Chart.yaml</strong><dd><code>Create Helm chart for AutoGPT server</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/Chart.yaml

- Created Helm chart for AutoGPT server.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-d1311f355a1fe4d2f7d72afd6f790ce61a81730789cbee561e7ebf75bc38e15b">+10/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>_helpers.tpl</strong><dd><code>Add helper templates for Helm chart</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/templates/_helpers.tpl

- Added helper templates for naming and labels.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-a02e792d11df7595be77ff275d853d5f22f25cc490a1421a88aac3e200867415">+62/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>configmap.yaml</strong><dd><code>Create ConfigMap template for environment variables</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/templates/configmap.yaml

- Created ConfigMap template for environment variables.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-df00517e09c991c0f6a4f47975e69885f8096aee3673c3e78f5f4a7dc771b55c">+10/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>deployment.yaml</strong><dd><code>Create Deployment template for AutoGPT server</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/templates/deployment.yaml

- Created Deployment template for AutoGPT server.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-1cc1e9e936903924fa06acca73b9fe72d1b2628445c0f49d5617e692bf7aa160">+80/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>hpa.yaml</strong><dd><code>Create HorizontalPodAutoscaler template</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/templates/hpa.yaml

- Created HorizontalPodAutoscaler template.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-fccb199cc8efcf7000859914a4503517ccdcb23b88b304efdc558b3c990e05bf">+32/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>ingress.yaml</strong><dd><code>Create Ingress template for AutoGPT server</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/templates/ingress.yaml

- Created Ingress template for AutoGPT server.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-4371b769f569c636a0d80489a685f250322844e62838e5cf6cc1af98968fd89b">+61/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>managedcert.yaml</strong><dd><code>Create ManagedCertificate template</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/templates/managedcert.yaml

- Created ManagedCertificate template.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-e4ca83e019ac67037f49faa131f42eb04c6ee8521ef791d0261b40b590036014">+7/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>service.yaml</strong><dd><code>Create Service template for AutoGPT server</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/templates/service.yaml

- Created Service template for AutoGPT server.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-e1aaa50bad888f0ef48627648dd667e6d9d5e9dbd6da53a22ef603353cef27aa">+19/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>serviceaccount.yaml</strong><dd><code>Create ServiceAccount template</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/templates/serviceaccount.yaml

- Created ServiceAccount template.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-a1abdd6b61ab71fdad740b57431201e3c90a32eedc80eaf3bc4089941e3063cc">+13/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>values.dev.yaml</strong><dd><code>Add development-specific values for Helm chart</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/values.dev.yaml

- Added development-specific values for Helm chart.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-c3ebccd239f6e53f74c6f8a1b70917df42fb1b6b1d573febe8d0ad21448db33d">+83/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>values.yaml</strong><dd><code>Add base values for Helm chart</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/values.yaml

- Added base values for Helm chart.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-d71710ad0f5eb834f9a0315c95897c3d38c3372a2f8303e51bff6fdc2580d3a6">+87/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>dev.tfvars</strong><dd><code>Add development environment-specific variables</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/terraform/environments/dev.tfvars

- Added development environment-specific variables.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-afbe1103c303a8001bc8c27865476c0e3d97707e560b5b023a627798504f5e38">+43/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Documentation</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>NOTES.txt</strong><dd><code>Add access instructions for Helm deployment</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/templates/NOTES.txt

- Added instructions for accessing the application.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-0fff085a4eff9c28062e7484897729e29842f928bccc07451a4b8af03a47b915">+22/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>test-connection.yaml</strong><dd><code>Create test Pod template for connection testing</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

rnd/infra/helm/autogpt-server/templates/tests/test-connection.yaml

- Created test Pod template for connection testing.

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7612/files#diff-51374db9afba183cbe54cacfb1f3e368d4bbda8bcf8add59ff05603b094593a7">+15/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: rnd/autogpt_server/autogpt_server/server/server.py [hunk 1]
```diff
@@ -183,12 +183,6 @@ def run_service(self):
 
         app.add_exception_handler(500, self.handle_internal_error)  # type: ignore
 
-        app.mount(
```
Comment (aarushik93) [inline rnd/autogpt_server/autogpt_server/server/server.py:194]: Not needed atm

File: rnd/infra/helm/autogpt-server/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            {{- toYaml .Values.livenessProbe | nindent 12 }}
+          readinessProbe:
+            {{- toYaml .Values.readinessProbe | nindent 12 }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          {{- with .Values.volumeMounts }}
+          volumeMounts:
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
+        - name: cloud-sql-proxy
```
Comment (aarushik93) [inline rnd/infra/helm/autogpt-server/templates/deployment.yaml:56]: Side car pattern for cloudsql

File: rnd/infra/helm/autogpt-server/values.dev.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  name: "dev-agpt-server-sa"
+
+service:
+  type: LoadBalancer
+  port: 80
+  targetPort: 8000
+
+ingress:
+  enabled: true
+  className: "gce"
```
Comment (aarushik93) [inline rnd/infra/helm/autogpt-server/values.dev.yaml:22]: Global External LB on GCP

File: rnd/infra/helm/autogpt-server/values.dev.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+serviceAccount:
+  annotations:
+    iam.gke.io/gcp-service-account: "dev-agpt-server-sa@agpt-dev.iam.gserviceaccount.com"
+  name: "dev-agpt-server-sa"
+
+service:
+  type: LoadBalancer
+  port: 80
+  targetPort: 8000
```
Comment (aarushik93) [inline rnd/infra/helm/autogpt-server/values.dev.yaml:16]: port = service port and target port = app port

File: rnd/infra/terraform/environments/dev.tfvars [hunk 5]
```diff
@@ -10,3 +10,37 @@ node_pool_name  = "dev-main-pool"
...
+
+workload_identity_bindings = {
+  "dev-agpt-server-workload-identity" = {
+    service_account_name = "dev-agpt-server-sa"
+    namespace            = "dev-agpt"
+    ksa_name             = "dev-agpt-server-sa"
+  }
+}
+
+role_bindings = {
```
Comment (aarushik93) [inline rnd/infra/terraform/environments/dev.tfvars:30]: common roles. add new roles as needed or add members as needed

File: rnd/infra/terraform/environments/dev.tfvars [hunk 6]
```diff
@@ -10,3 +10,37 @@ node_pool_name  = "dev-main-pool"
...
+
+
+service_accounts = {
+  "dev-agpt-server-sa" = {
+    display_name = "AutoGPT Dev Server Account"
+    description  = "Service account for agpt dev server"
+  }
+}
+
+workload_identity_bindings = {
```
Comment (aarushik93) [inline rnd/infra/terraform/environments/dev.tfvars:22]: Workload identity so a kubernetes account can act with the same perms as a GCP service account, _without_ having to pass around credentials as json files

**Comment by aarushik93** [comment]
This PR Is a bit large so I can break it into 2 if you prefer for reviews

File: rnd/autogpt_server/autogpt_server/util/service.py [hunk 7]
```diff
@@ -16,6 +17,8 @@
 conn_retry = retry(stop=stop_after_delay(5), wait=wait_exponential(multiplier=0.1))
 T = TypeVar("T")
 
+host = os.environ.get("PYRO_HOST", "localhost")
```
Comment (ntindle) [inline rnd/autogpt_server/autogpt_server/util/service.py]: Read from config `autogpt_server/util/settings.py` so we have a consistent interface for configurables

**Review by ntindle — CHANGES_REQUESTED** [review]

File: rnd/autogpt_server/autogpt_server/util/service.py (hunk 7 above)
Comment (aarushik93) [reply rnd/autogpt_server/autogpt_server/util/service.py]: @ntindle I tried to get it to read from Config, but it doesn't read the .env it seems. Only the config.json. So I added it to Secrets which does read from .env. Is there a way to make Config work to read from .env?

Comment (aarushik93) [reply rnd/autogpt_server/autogpt_server/util/service.py]: updated to use Config, since bug fix came in!

**Review by ntindle — APPROVED** [review]

**Pull request merged**